addons/survey/controllers/main.py

```diff
@@ -148,17 +142,15 @@ def fill_survey(self, survey, token, prev=None, **post):
         else:
             return request.render("survey.403", {'survey': survey})
 
-    # AJAX prefilling of a survey
-    @http.route(['/survey/prefill/<model("survey.survey"):survey>/<string:token>',
-                 '/survey/prefill/<model("survey.survey"):survey>/<string:token>/<model("survey.page"):page>'],
-                type='http', auth='public', website=True)
-    def prefill(self, survey, token, page=None, **post):
+    @http.route('/survey/prefill/<model("survey.survey"):survey>/<string:token>', type='http', auth='public', website=True)
```

Comment:
Yes. Some routes are strange, called by a fake form and managed by some JS code. It should probably be JSONified. This will be done in a near future. Here I don't change those routes, just fix some bits of code.